AE_PunktBeschriftung/include/representations/Solution.hpp

```diff
@@ -1,32 +1,58 @@
-#pragma once
+#ifndef LABEL_MAP_SOLUTION_HPP
+#define LABEL_MAP_SOLUTION_HPP
 
-#include <representations/Instance.hpp>
-#include <representations/Point.hpp>
 
-#include <map>
+#include "Instance.hpp"
 
 class Solution {
-
-private:
-
-    const Instance& instance;
-
-    std::map<int, Point::Rectangle> placements;
-
-    friend std::ostream& operator<<(std::ostream &ostream, const Solution &solution);
+protected:
+    const Instance instance;
 
 public:
-
-    Solution(const Instance& instance);
-
-    void setLabel(int idx, Point::Corner corner);
-
-    bool contains(int idx) const;
-
-    bool checkCollision(const Point& p, Point::Corner placement, int otherIdx) const;
-
-    int size() const;
-
-    bool isFeasible() const;
-
-};
\ No newline at end of file
+    /**
+     * Operator to print the solution to something. Must be called like: solution << ostream;
+     * Its the only way i can think of to require a printable solution via << operator.
+     *
+     * @param ostream the stream to be printed to
+     * @return the out stream from the input
+     */
+    virtual std::ostream &operator<<(std::ostream &ostream) = 0;
```

Comment:
I don't think this is a good way of doing that - you would use this operator like std::cout << solution << std::cout; right? That just does not make sense, what is the point of overloading the operator then? If this really is not possible, we should at least do it as a print() method or something, because the whole point of operator overloading is to make the code more readable